### ISSUE TITLE
Need a parameter to launch scrcpy without console

### ISSUE THREAD

**Issue by Mogez** [Genymobile/scrcpy#2017]
VBS works but how about add a command line switch? It'll be clean and clear.

Thanks a lot for your work!

**Comment by rom1v** [comment]
That would be great, but that's not possible in the scrcpy executable itself: the fact that there is a console or no console is determined before running the executable.

**Comment by Mogez** [comment]
> That would be great, but that's not possible in the scrcpy executable itself: the fact that there is a console or no console is determined before running the executable.

I see. I wish I could make a scrcpy-noconsole.exe like the previous versions...

**Comment by rom1v** [comment]
You could probably modify `scrcpy-noconsole.vbs` to forward the command-line arguments, so `scrcpy-noconsole.vbs` would behave the same way as the old `scrcpy-noconsole.exe`.

**Comment by Mogez** [comment]
Yeah, I finally made a legacy VB6 starter app to pass it on, it can be added to batch file in one line like this:

> adb connect 192.168.1.24:5555
> **_starter_** scrcpy.exe _**--noconsole**_ --disable-screensaver --max-fps 60 -m 1440 --push-target "/sdcard/download/" --shortcut-mod lctrl -S -w --window-title "Android Rocks!" -s 192.168.1.24:5555

**Comment by rom1v** [comment]
> `scrcpy.exe --noconsole ...`

What does it do? Please show the diff.

**Comment by livingator** [comment]
> 
> 
> You could probably modify `scrcpy-noconsole.vbs` to forward the command-line arguments, so `scrcpy-noconsole.vbs` would behave the same way as the old `scrcpy-noconsole.exe`.

can i know how to do it? i prefer not to have console opened usually if it's possible

**Comment by chavocarlos** [comment]
> 
> 
> > You could probably modify `scrcpy-noconsole.vbs` to forward the command-line arguments, so `scrcpy-noconsole.vbs` would behave the same way as the old `scrcpy-noconsole.exe`.
> 
> ~can i know how to do it? i prefer not to have console opened usually if it's possible~
> 
> nevermind, figured out how to do it

Can you share with us, please?

**Comment by livingator** [comment]
> 
> 
> > > You could probably modify `scrcpy-noconsole.vbs` to forward the command-line arguments, so `scrcpy-noconsole.vbs` would behave the same way as the old `scrcpy-noconsole.exe`.
> > 
> > 
> > ~can i know how to do it? i prefer not to have console opened usually if it's possible~
> > nevermind, figured out how to do it
> 
> Can you share with us, please?

edit the .vbs file with notepad or anything you prefer and add your arguments after scrcpy.exe and before the end of double quotation.
below is my.vbs for example
`CreateObject("Wscript.Shell").Run "cmd /c scrcpy.exe --stay-awake --render-driver=opengl --shortcut-mod=rctrl", 0, false
`
hope it helps

**Comment by chavocarlos** [comment]
> 
> 
> > > > You could probably modify `scrcpy-noconsole.vbs` to forward the command-line arguments, so `scrcpy-noconsole.vbs` would behave the same way as the old `scrcpy-noconsole.exe`.
> > > 
> > > 
> > > ~can i know how to do it? i prefer not to have console opened usually if it's possible~
> > > nevermind, figured out how to do it
> > 
> > 
> > Can you share with us, please?
> 
> edit the .vbs file with notepad or anything you prefer and add your arguments after scrcpy.exe and before the end of double quotation.
> below is my.vbs for example
> `CreateObject("Wscript.Shell").Run "cmd /c scrcpy.exe --stay-awake --render-driver=opengl --shortcut-mod=rctrl", 0, false `
> hope it helps

I swear that already did it, but think I did some mistake, now works. Thanks a lot.

**Comment by luronumen** [comment]
`scrcpy-noconsole.vbs` must supports parameters because if multiples devices are connected with the PC we have to identify what cellphone we want to connect by `--serial` parameter

**Comment by rom1v** [comment]
https://github.com/Genymobile/scrcpy/pull/2052#issuecomment-765562412

**Comment by luronumen** [comment]
Thanks @rom1v

**Comment by BlueStarBRS** [comment]
i still want --noconsole and not a god damm ugly vbs that i cant pin to my start menu im sick of convert my vbs scripts into exe files who are over and over get delted by antivirus for nothing for real ngl

**Comment by AlphaCorpDESIGN** [comment]
Hello everyone, I'm not sure if this is still relevant, but if you're still looking, add these lines at the beginning of the .bat file to automatically minimize the console window after launching.

@echo off
if not "%minimized%"=="" goto :minimized
set minimized=true
start /min cmd /C "%~dpnx0"
exit
:minimized

**Comment by Lopou94** [comment]
> Hello everyone, I'm not sure if this is still relevant, but if you're still looking, add these lines at the beginning of the .bat file to automatically minimize the console window after launching.
> 
> @echo off if not "%minimized%"=="" goto :minimized set minimized=true start /min cmd /C "%~dpnx0" exit :minimized

Hello

I followed your instructions and used the commands in the bat file but I still have 2 windows open.

The console windows will open behind the main scrcpy window even though I run the commands in bat

Any ideas??
![2024-08-12 19_33_07-pythonProject – ADBD py](https://github.com/user-attachments/assets/4821ac99-5ed4-4f7f-85f2-10c631a200e2)

![image](https://github.com/user-attachments/assets/9de41962-c029-404b-be3f-b741b99ee6c7)

Thanks in advance,

**Comment by rom1v** [comment]
Une `scrcpy-noconsole.vbs` if you don't want a console on Windows.

**Comment by Lopou94** [comment]
> Une `scrcpy-noconsole.vbs` if you don't want a console on Windows.

Yeah i found this file later on the folder.

I'm trying to run from python but I'm getting some errors

Thanks,

**Comment by rom1v** [comment]
If you run from python, don't use the .bat or .vbs, just exec `scrcpy.exe` with the arguments: https://stackoverflow.com/questions/89228/how-do-i-execute-a-program-or-call-a-system-command

**Comment by Lopou94** [comment]
> If you run from python, don't use the .bat or .vbs, just exec `scrcpy.exe` with the arguments you want: https://stackoverflow.com/questions/89228/how-do-i-execute-a-program-or-call-a-system-command

Yeah, i found the link that you sent but every time i running scrcpy from my software a console window opens directly if connection with smartphone is successful.

Can i use the arguments from the vbs when i am running the scrcpy.exe from my code?

Thanks,

**Comment by tomyerkes** [comment]
You can use a program called Bat to EXE converter and choose an option to make the console invisible. This will also allow you to then pin your exe to the start menu whereas you cannot pin .bat files.  I use this to have app icons for all my android apps that open in a new-display with scrcpy and then added a command to the end to close out the app when I close the window so it doesnt leave it hanging in my phone. Also I choose playback for audio when using media apps and display on screen keyboard.  Using a pixel this is the best way I could get back to my dex days with Samsung :/ 

![Image](https://github.com/user-attachments/assets/08b21ba9-9e03-4b5d-b10c-3f0605a1d83d)
